src/components/shared/Fields/Form/HookForm.tsx

```diff
@@ -40,7 +43,25 @@ const HookForm = <FormData extends FieldValues>({
     mode,
     ...options,
   });
-  const { handleSubmit } = formHelpers;
+
+  const {
+    handleSubmit,
+    watch,
+    reset,
+    formState: { isSubmitting },
+  } = formHelpers;
+  const values = watch();
+
+  /*
+   * Effect resets default values to latest values, which resets the isDirty prop.
+   * Useful in user settings.
+   */
+  useEffect(() => {
```

Comment:
This is so that when you hit save, the form re-disables, which it should because technically the form is now not dirty any longer.